dissect/target/plugins/scrape/scrape.py

```diff
@@ -204,3 +286,96 @@ def scrape_needles_from_disks(
             disk.seek(0)
             for needle, offset, match in find_needles(disk, needles=needles, block_size=block_size):
                 yield disk, needle, offset, match
+
+    @internal
+    def _get_disk_vols(self, disk: Container) -> Iterator[Volume]:
+        """Yields all volumes for a given disk container.
+
+        When the disk has an associated volume system, volumes are retrieved from there.
+        When the disk is raw, volumes are retrieved from the target's volume list.
+
+        Args:
+            disk: The disk container to get volumes for.
+        """
+        if disk.vs:
+            yield from sorted(disk.vs.volumes, key=lambda v: v.offset)
+        else:
+            for volume in sorted(self.target.volumes, key=lambda v: v.offset or 0):
+                if volume.disk == disk and not volume.vs:
+                    yield volume
+
+
+def process_lvm_layer(context: ScrapeContext, volume: Volume) -> bool:
+    if not (context.lvm_flag and isinstance(volume.vs, LogicalVolumeSystem)):
+        return False
+
+    dependencies_met = True
+    backing_regions_info = []
+
+    try:
+        for source_dev in volume.vs.fh:
+            backing_vol_obj = source_dev.fh
+
+            if backing_vol_obj not in context.volume_region_map:
+                dependencies_met = False
+                break
+
+            backing_regions_info.append(context.volume_region_map[backing_vol_obj])
+
+    except (AttributeError, TypeError):
+        dependencies_met = False
+
+    if dependencies_met:
```

Comment:
```suggestion
    if not dependencies_met:
        return False
```
if you invert the condition here, you can reduce the indentation down below. 
Tho you can also invert the `dependencies_met` logic and rename it as `dependencies_not_met` it depends in what reads better